sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -40,7 +42,16 @@
     private static final int MAX_BODY_LOG_SIZE = 1024 * 16;
     private static final String REDACTED_PLACEHOLDER = "REDACTED";
 
-    private final ClientLogger logger = new ClientLogger(HttpLoggingPolicy.class);
+    // Use a cache to retain the caller method ClientLogger.
+    //
+    // The same method may be called thousands or millions of times, so it is wasteful to create a new logger instance
+    // each time the method is called. Instead, retain the created ClientLogger until a certain number of unique method
+    // calls have been made and then clear the cache and rebuild it. Long term, this should be replaced with an LRU,
+    // or another type of cache, for better cache management.
+    private static final int LOGGER_CACHE_MAX_SIZE = 1000;
+    private static final Map<String, ClientLogger> CALLER_METHOD_LOGGER_CACHE = new ConcurrentHashMap<>();
```

Comment:
would it make sense to put dynamic part into log entry context instead of logger name and avoid cache of loggers? Or do we consider logger name change to be breaking?

Reply:
Not completely certain on policy but to me changing the logger name would be breaking. If I understand correct, using context you'd have the `HttpLoggingPolicy` logger do the logging with having the method name as a context property like `method: <the method>`, right? If so a customer who set up `<the method name package>: ERROR` and `com.azure.core: INFO` would then have their logging output changed. So, unfortunately, caching of this kind would be needed to retain current behavior and to optimize performance.

Reply:
i agree it might be somewhat breaking, but there is no means or practice in the industry to enforce stability of log messages content, level, presence, order, or where they are written from. When we refactor something and shuffle classes around, we don't try to preserve logger behavior/name - it would be impossible.
It feels like concurrent logger cache could be a worse evil than changing logger name here.